src/ray/core_worker/core_worker.cc

```diff
@@ -0,0 +1,38 @@
+#include "core_worker.h"
+#include "context.h"
+
+namespace ray {
+
+CoreWorker::CoreWorker(const enum WorkerType worker_type, const enum Language language,
+                       const std::string &store_socket, const std::string &raylet_socket,
+                       DriverID driver_id)
+    : worker_type_(worker_type),
+      language_(language),
+      worker_context_(worker_type, driver_id),
+      store_socket_(store_socket),
+      raylet_socket_(raylet_socket),
+      task_interface_(*this),
+      object_interface_(*this),
+      task_execution_interface_(*this) {}
+
+Status CoreWorker::Connect() {
+  // connect to plasma.
+  RAY_ARROW_RETURN_NOT_OK(store_client_.Connect(store_socket_));
+
+  // connect to raylet.
+  ::Language lang = ::Language::PYTHON;
+  if (language_ == Language::JAVA) {
+    lang = ::Language::JAVA;
+  }
```

Comment:
Why not `::Language lang = language_ ` simply ?

Reply:
the assignment cannot be done directly as they are different types - the former comes from flatbuffer, while the latter is defined in core worker.